power_grid_model_c/power_grid_model_c/include/power_grid_model_c/dataset.h

```diff
@@ -83,6 +83,47 @@ PGM_API PGM_Idx PGM_dataset_info_elements_per_scenario(PGM_Handle* handle, PGM_D
  */
 PGM_API PGM_Idx PGM_dataset_info_total_elements(PGM_Handle* handle, PGM_DatasetInfo const* info, PGM_Idx component_idx);
 
+/**
+ * @brief Return if a component has attribute indications.
+ *
+ * Attribute indications are used to indicate the presence of meaningful attributes for a certain component in the
+ * dataset. It returns 1 for a component if all the elements of all the scenarios for this component are list-like in
+ * the serialized data.
+ *
+ * @param handle
+ * @param info A pointer to the info object.
+ * @param component_idx The index number of the component.
+ * @return 1 if the component has attribute indications, 0 if it does not.
+ */
+PGM_API PGM_Idx PGM_dataset_info_has_attribute_indications(PGM_Handle* handle, PGM_DatasetInfo const* info,
+                                                           PGM_Idx component_idx);
+
+/**
+ * @brief Return the number of attribute indications for a component.s
+ *
+ * @param handle
+ * @param info A pointer to the info object.
+ * @param component_idx The index number of the component.
+ * @return The number of attribute indications for the component.
+ * It is UB if PGM_dataset_info_has_attribute_indications() returns zero.
```

Comment:
why is this necessary? it may just return `0`, right?

Reply:
Why is the UB necessary?

Reply:
It has been changed.